### PR TITLE
Feature/wallet data

### PR DESCRIPTION
**Pull request by matthewcarlreetz**

### CODE REVIEW

File: src/features/wallet/root/ActivityItem.tsx [hunk 1]
```diff
@@ -1,49 +1,175 @@
...
   isLast = false,
+  address = '',
 }: Props) => {
-  const handlePress = () => {
-    triggerNavHaptic()
-  }
+  const colors = useColors()
+  const { t } = useTranslation()
+
+  const isSending = useMemo(() => isPayer(address, item), [address, item])
```
Comment (tyler-whitman) [inline src/features/wallet/root/ActivityItem.tsx:54]: Something is up with this. Mine say Received but they are actually Sent payments

Comment (tyler-whitman) [reply src/features/wallet/root/ActivityItem.tsx:54]: Actually it looks like its always false

File: src/features/wallet/root/ActivityItem.tsx [hunk 2]
```diff
@@ -1,49 +1,175 @@
...
+
+  const icon = useCallback(
+    (type: string) => {
+      if (!TxnTypeKeys.find((k) => k === type)) {
+        return null
+      }
+
+      const size = 24
+      switch (type as TxnType) {
+        case 'transfer_hotspot_v1':
```
Comment (tyler-whitman) [inline src/features/wallet/root/ActivityItem.tsx:152]: Should we save these types in an enum or something somewhere? Or do they type match already?

Comment (matthewcarlreetz) [reply src/features/wallet/root/ActivityItem.tsx:152]: TxnType gives us type safety here.

File: src/features/wallet/root/ActivityItem.tsx (hunk 1 above)
Comment (matthewcarlreetz) [reply src/features/wallet/root/ActivityItem.tsx:54]: ok, I'm going to try some other addresses and see if I can replicate.

File: src/App.tsx [hunk 3]
```diff
@@ -41,30 +42,37 @@ const App = () => {
       isRestored,
       isBackedUp,
       isRequestingPermission,
+      isLocked,
+      appStateStatus,
     },
   } = useSelector((state: RootState) => state)
 
+  useEffect(() => {
```
Comment (tyler-whitman) [inline src/App.tsx:50]: Does this fix something? Just wondering. I see we moved from using the direct callback to using the state and effect instead. Makes sense just wondering if it caused a bug or something?

File: src/App.tsx [hunk 4]
```diff
@@ -73,6 +81,12 @@ const App = () => {
     dispatch(fetchData())
   }, [isRestored, isBackedUp, dispatch])
 
+  useEffect(() => {
+    if (isBackedUp) {
```
Comment (tyler-whitman) [inline src/App.tsx:85]: Whats the flow on this? It seems like something that should happen once on launch of the app? 

Why do we do it only if `isBackedUp` is true? And we need to run again if it goes false and back to true. Im wondering if just passing [] would work?

File: src/App.tsx (hunk 3 above)
Comment (matthewcarlreetz) [reply src/App.tsx:50]: There was a bug on android where face id would send the app into 'background' which was causing an infinite loop of lock=>auth=>lock=>auth

File: src/App.tsx (hunk 4 above)
Comment (matthewcarlreetz) [reply src/App.tsx:85]: We need to wait for it to be backed up so that the account address is defined. `client.account(address).activity.list()` requires account, so we reinitialize them whenever `isBacked` changes to true.

File: src/features/wallet/root/ActivityCard.tsx [hunk 5]
```diff
@@ -34,51 +88,75 @@ const ActivityCard = forwardRef(
...
+            if (isPendingTransaction(item)) {
+              return `${filter}.${(item as PendingTransaction).hash}`
+            }
+
+            return `${filter}.${(item as AddGatewayV1).hash}`
+          },
           renderItem,
+          refreshControl: (
+            <RefreshControl
+              refreshing={txnStatus === 'pending' && !transactionData.length}
```
Comment (tyler-whitman) [inline src/features/wallet/root/ActivityCard.tsx:150]: I dont see this when switching filters as it loads. I like the animation but I think it needs some kind of loading indicator or it feels like its frozen for a bit when loading transactions after changing filters.

Maybe we can add another boolean when the filter changes and its loading? We could also do this in a separate PR if you want as it works well already.

File: src/features/wallet/root/ActivityCardHeader.tsx [hunk 6]
```diff
@@ -1,32 +1,63 @@
...
 
-const ActivityCardHeader = () => {
-  const handlePress = () => {
-    triggerNavHaptic()
+type Props = {
+  filter: FilterType
+  onFilterChanged: (filter: FilterType) => void
+}
+
+const ActivityCardHeader = ({ filter, onFilterChanged }: Props) => {
```
Comment (tyler-whitman) [inline src/features/wallet/root/ActivityCardHeader.tsx:17]: Not sure if it would be a lot of work, but we will need this in other places too so making it generic would be awesome.

Maybe instead of using specific filters we could make it a generic string array? We could also change it if we want when we need to use it in another place?

File: src/features/wallet/root/ActivityItem.tsx (hunk 1 above)
Comment (tyler-whitman) [reply src/features/wallet/root/ActivityItem.tsx:54]: Its fixed now

File: src/features/wallet/root/ActivityItem.tsx (hunk 2 above)
Comment (tyler-whitman) [reply src/features/wallet/root/ActivityItem.tsx:159]: Cool, works for me

File: src/features/wallet/root/ActivityItem.tsx [hunk 7]
```diff
@@ -63,34 +196,38 @@ const ActivityItem = ({
...
-          </Text>
+          {time && <Text color="graySteel">{time}</Text>}
         </Box>
       </Box>
     </TouchableWithoutFeedback>
   )
 }
 
-export default ActivityItem
+export default memo(ActivityItem)
```
Comment (tyler-whitman) [inline src/features/wallet/root/ActivityItem.tsx:233]: Nice, I think we should be using this for most future components?

File: src/App.tsx (hunk 3 above)
Comment (tyler-whitman) [reply src/App.tsx:50]: Ah nice good find

File: src/App.tsx (hunk 4 above)
Comment (tyler-whitman) [reply src/App.tsx:85]: Ah ok, that makes sense

**Review by tyler-whitman — APPROVED** [review]

File: src/features/wallet/root/ActivityItem.tsx (hunk 7 above)
Comment (matthewcarlreetz) [reply src/features/wallet/root/ActivityItem.tsx:233]: definitely for flatlist items

File: src/features/wallet/root/ActivityCard.tsx (hunk 5 above)
Comment (matthewcarlreetz) [reply src/features/wallet/root/ActivityCard.tsx:150]: hmmm. Are you testing on ios or android?

File: src/features/wallet/root/ActivityCardHeader.tsx (hunk 6 above)
Comment (matthewcarlreetz) [reply src/features/wallet/root/ActivityCardHeader.tsx:17]: yeah, I'm with you on that. For now, I'd rather leave it as a todo

File: src/features/wallet/root/ActivityCard.tsx (hunk 5 above)
Comment (tyler-whitman) [reply src/features/wallet/root/ActivityCard.tsx:150]: The iOS simulator

File: src/features/wallet/root/ActivityCardHeader.tsx (hunk 6 above)
Comment (tyler-whitman) [reply src/features/wallet/root/ActivityCardHeader.tsx:17]: Yeah sounds good
